packages/flutter/lib/src/cupertino/app.dart

```diff
@@ -268,47 +269,376 @@ class _CupertinoAppState extends State<CupertinoApp> {
+              showPerformanceOverlay: widget.showPerformanceOverlay,
+              checkerboardRasterCacheImages: widget.checkerboardRasterCacheImages,
+              checkerboardOffscreenLayers: widget.checkerboardOffscreenLayers,
+              showSemanticsDebugger: widget.showSemanticsDebugger,
+              debugShowCheckedModeBanner: widget.debugShowCheckedModeBanner,
+              inspectorSelectButtonBuilder: (BuildContext context, VoidCallback onPressed) {
+                return CupertinoButton.filled(
+                  child: const Icon(
+                    CupertinoIcons.search,
+                    size: 28.0,
+                    color: CupertinoColors.white,
+                  ),
+                  padding: EdgeInsets.zero,
+                  onPressed: onPressed,
+                );
+              },
+            ),
+          ),
         ),
       ),
     );
   }
 }
+
+/// Establishes a subtree where iOS system colors resolve to the given data.
+///
+/// Typically the given [CupertinoSystemColorsData] is resolved against its own
+/// [BuildContext] using [CupertinoSystemColorsData.resolveColors].
+class CupertinoSystemColors extends InheritedWidget {
+  const CupertinoSystemColors._({
+    Key key,
+    @required CupertinoSystemColorsData data,
+    Widget child,
+  }) : _data = data,
+       assert(data != null),
+       super(key: key, child: child);
+
+  /// Creates a widget that provides a [CupertinoSystemColorsData] which is extracted
+  /// from the given [BuildContext] to its descendants.
+  ///
+  /// The [context] and [child] arguments must not be null.
+  CupertinoSystemColors.fromCurrentContext({
+    Key key,
+    BuildContext context,
+    Widget child,
+  }) : this._(key: key, data: CupertinoSystemColors.of(context).resolveColors(context), child: child);
+
+  final CupertinoSystemColorsData _data;
+
+  @override
+  bool updateShouldNotify(CupertinoSystemColors oldWidget) => oldWidget._data != _data;
+
+
+  /// Retrieves the iOS system colors from the given [BuildContext].
+  ///
+  /// Falls back to [fromSystem] and then [fallbackValues], if such information
+  /// is not available in the given [BuildContext].
+  static CupertinoSystemColorsData of(BuildContext context) {
+    final CupertinoSystemColors widget = context.inheritFromWidgetOfExactType(CupertinoSystemColors);
+    return widget?._data ?? fromSystem ?? fallbackValues;
+  }
+
+  /// System Colors fetched from the OS.
+  ///
+  /// On iOS 13 and above all the colors will be fetched from the operating system.
+  /// On Android and lower versions of iOS this will return null.
+  static CupertinoSystemColorsData get fromSystem => null;
+
+  /// Fallback System Colors, extracted from:
+  /// https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/color/#dynamic-system-colors
+  /// and iOS 13 beta.
+  static CupertinoSystemColorsData get fallbackValues {
+    return CupertinoSystemColorsData(
+      label: CupertinoDynamicColor(
```

Comment:
Unfortunately `CupertinoDynamicColor`'s constructors can't be made const because of calling `super(value.value)` is not allowed in a const constructor (more specifically, using the `Color.value` getter).

Reply:
Ah got it, thanks for the explanation.